### PR TITLE
commands/operator-sdk/cmd: Add optional path to operator.yaml template

### PR DESCRIPTION
**Pull request by theishshah**
In reference to discussions had on #84 this allows for a custom operator.yaml template to given to the `operator-sdk build` command

### CODE REVIEW

**Comment by hasbro17** [comment]
@theishshah Can you provide an example of how this works and what the input template file looks like.

**Comment by fanminshi** [comment]
@theishshah agree with @hasbro17. Could you provide a simple design on #84 on how are you going to solve this issue before we start to code.

**Comment by theishshah** [comment]
The tool now works such that if a operator.yaml exists no new file is generated. This allows either a fully custom file to be created before the build command, or for a file to be modified in between builds without being over-written. If the image name changes between builds it is the user responsibilty to update that in the file. cc @hasbro17 @fanminshi

File: commands/operator-sdk/cmd/new.go [hunk 1]
```diff
@@ -86,6 +86,19 @@ func newFunc(cmd *cobra.Command, args []string) {
 	}
 	pullDep()
 	generate.K8sCodegen(projectName)
+	// c := &generator.Config{}
```
Comment (fanminshi) [inline commands/operator-sdk/cmd/new.go]: if code is not being used, we should get rid of it?

**Comment by fanminshi** [comment]
also update the change.log to reflect this change.

File: doc/user-guide.md [hunk 2]
```diff
@@ -114,6 +114,7 @@ Run as pod inside a Kubernetes cluster is preferred for production use.
 Build the memcached-operator image and push it to a registry:
 ```
 $ operator-sdk build quay.io/example/memcached-operator:v0.0.1
+$ sed -i 's/REPLACE_IMAGE/quay.io\/example\/memcached-operator:v0.0.1/g' deploy/operator.yaml
```
Comment (fanminshi) [inline doc/user-guide.md]: i think we also need to reflect the `readme.md` for this change.

File: commands/operator-sdk/cmd/new.go (hunk 1 above)
Comment (theishshah) [reply commands/operator-sdk/cmd/new.go]: Yes, it is currently commented out to test on travis, if it works I will remove the code

File: README.md [hunk 3]
```diff
@@ -63,6 +63,7 @@ $ cd app-operator
 
 # Build and push the app-operator image to a public registry such as quay.io
 $ operator-sdk build quay.io/example/app-operator
+$ sed -i 's/REPLACE_IMAGE/quay.io\/example\/app-operator/g' deploy/operator.yaml
```
Comment (fanminshi) [inline README.md]: let's add a comment here to explain what's going on.

**Comment by fanminshi** [comment]
lgtm

File: README.md [hunk 4]
```diff
@@ -62,7 +62,10 @@ $ operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=Ap
 $ cd app-operator
 
 # Build and push the app-operator image to a public registry such as quay.io
+# We use sed to replace the placeholder image name that is generated the creation point of the operator
+# The sed operation only needs to be down once during the first build
 $ operator-sdk build quay.io/example/app-operator
+$ sed -i 's/REPLACE_IMAGE/quay.io\/example\/app-operator/g' deploy/operator.yaml
```
Comment (hasbro17) [inline README.md]: Can you use a different separator so we don't have to escape `/` which makes the image name confusing e.g
`sed -i 's|REPLACE_IMAGE|quay.io/example/app-operator|g'`

File: README.md [hunk 5]
```diff
@@ -62,7 +62,10 @@ $ operator-sdk new app-operator --api-version=app.example.com/v1alpha1 --kind=Ap
 $ cd app-operator
 
 # Build and push the app-operator image to a public registry such as quay.io
+# We use sed to replace the placeholder image name that is generated the creation point of the operator
+# The sed operation only needs to be down once during the first build
 $ operator-sdk build quay.io/example/app-operator
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/app-operator|g' deploy/operator.yaml
 $ docker push quay.io/example/app-operator
```
Comment (hasbro17) [inline README.md]: We can rearrange the steps and make the above two lines more concise:
```
# Build and push the app-operator image to a public registry such as quay.io
$ operator-sdk build quay.io/example/app-operator
$ docker push quay.io/example/app-operator

# Update the operator manifest to use the built image name
$ sed -i 's|REPLACE_IMAGE|quay.io/example/app-operator|g' deploy/operator.yaml
```

**Pull request merged**